fronts-client/src/services/recipeQuery.ts

```diff
@@ -0,0 +1,167 @@
+import url from '../constants/url';
+import { Recipe } from '../types/Recipe';
+
+interface KeyAndCount {
+  key: string;
+  doc_count: number;
+}
+
+export interface ChefSearchParams {
+  query?: string;
+  limit?: number;
+}
+
+export interface RecipeSearchFilters {
+  diets?: string[];
+  contributors?: string[];
+  filterType: "During" | "Post";
+}
+
+export interface RecipeSearchParams {
+  queryText: string;
+  searchType?: "Embedded"|"Match"|"Lucene",
+  fields?: string[];
+  kfactor?: number;
+  limit?: number;
+  filters?: RecipeSearchFilters;
+}
+
+export interface ChefSearchResponse {
+
+}
+
+export interface RecipeSearchResponse {
+  hits: number;
+  maxScore: number;
+  recipes: RecipeSearchHit[];
+}
+
+interface RecipeSearchTitlesResponse {
+  score: number;
+  href: string;
+}
+
+export type RecipeSearchHit = Recipe & {
+  score: number;
+}
+
+export interface DietSearchResponse {
+  "diet-ids": KeyAndCount[];
+}
+
+const widthParam = /width=(\d+)/;
+export const updateImageScalingParams = (url:string) => {
+  return url.replace(widthParam, "width=83")
+}
+
+const setupRecipeThumbnails = (recep:Recipe) => {
```

Comment:
I gotta ask. This looks like a mispelling, but you're consistent with it. rec-E-P? As in, phonetic? 😅

Reply:
Yeah it's just an abbrieviation that I like to use, as you know I type fast and it's really easy to mis-spell "recipe" as well as it being longer so for code I prefer `recep` (`ress - ep`) as it still conveys the meaning (I think!) and slows me down less.